File: NeuSoftTrainingProject/backend/bsp/src/main/java/com/neusoft/bsp/codemaster/entity/Cdm.java
package com.neusoft.bsp.codemaster.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import org.springframework.format.annotation.DateTimeFormat;

import java.sql.Timestamp;

public class Cdm {

    int cdm_id;
    String code_type;
    String description;
    String type_cd;
    String code_val;
    int seq_no;
    String created_by;
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    Timestamp creation_date;
    String last_update_by;
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    Timestamp last_update_date;
    int call_cnt;
    String remark;
    int sts_cd;

    public int getCdm_id() {
        return cdm_id;
    }

    public void setCdm_id(int cdm_id) {
        this.cdm_id = cdm_id;
    }

    public String getCode_type() {
        return code_type;
    }

    public void setCode_type(String code_type) {
        this.code_type = code_type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getType_cd() {
        return type_cd;
    }

    public void setType_cd(String type_cd) {
        this.type_cd = type_cd;
    }

    public String getCode_val() {
        return code_val;
    }

    public void setCode_val(String code_val) {
        this.code_val = code_val;
    }

    public int getSeq_no() {
        return seq_no;
    }

    public void setSeq_no(int seq_no) {
        this.seq_no = seq_no;
    }

    public String getCreated_by() {
        return created_by;
    }

    public void setCreated_by(String created_by) {
        this.created_by = created_by;
    }

    public Timestamp getCreation_date() {
        return creation_date;
    }

    public void setCreation_date(Timestamp creation_date) {
        this.creation_date = creation_date;
    }

    public String getLast_update_by() {
        return last_update_by;
    }

    public void setLast_update_by(String last_update_by) {
        this.last_update_by = last_update_by;
    }

    public Timestamp getLast_update_date() {
        return last_update_date;
    }

    public void setLast_update_date(Timestamp last_update_date) {
        this.last_update_date = last_update_date;
    }

    public int getCall_cnt() {
        return call_cnt;
    }

    public void setCall_cnt(int call_cnt) {
        this.call_cnt = call_cnt;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

    public int getSts_cd() {
        return sts_cd;
    }

    public void setSts_cd(int sts_cd) {
        this.sts_cd = sts_cd;
    }

    @Override
    public String toString() {
        return "Cdm{" +
                "cdm_id=" + cdm_id +
                ", code_type='" + code_type + '\'' +
                ", description='" + description + '\'' +
                ", type_cd='" + type_cd + '\'' +
                ", code_val='" + code_val + '\'' +
                ", seq_no=" + seq_no +
                ", created_by='" + created_by + '\'' +
                ", creation_date=" + creation_date +
                ", last_update_by='" + last_update_by + '\'' +
                ", last_update_date=" + last_update_date +
                ", call_cnt=" + call_cnt +
                ", remark='" + remark + '\'' +
                ", sts_cd=" + sts_cd +
                '}';
    }
}

File: NeuSoftTrainingProject/frontend/vue-element-admin/src/store/modules/cdm.js
import {getCdms,searchCdm,insertCdm,updateCdm,deleteCdm} from "../../api/cdmHandler";

const cdm = {
  actions : {

    GetCdms({ commit }){
      return new Promise((resolve,reject) => {
        getCdms().then(response => {
          // commit('')
          resolve(response)
        }).catch(error => {
          reject(error)
        })
      })
    },

    SearchCdm({ commit },code_type){
      return new Promise((resolve,reject) => {
        searchCdm(code_type).then(response => {
          // commit('')
          resolve(response)
        }).catch(error => {
          reject(error)
        })
      })
    },

    InsertCdm({ commit },cdmInfo){
      return new Promise((resolve,reject) => {
        insertCdm(cdmInfo.code_type,cdmInfo.description,cdmInfo.type_cd,cdmInfo.code_val,cdmInfo.seq_no,
                  cdmInfo.created_by,cdmInfo.creation_date,cdmInfo.last_update_by,cdmInfo.last_update_date,
                  cdmInfo.call_cnt,cdmInfo.remark,cdmInfo.sts_cd).then(response => {
          // commit('')
          resolve(response)
        }).catch(error => {
          reject(error)
        })
      })
    },

    UpdateCdm({ commit },cdmInfo){
      return new Promise((resolve,reject) => {
        updateCdm(cdmInfo.cdm_id,cdmInfo.code_type,cdmInfo.description,cdmInfo.type_cd,cdmInfo.code_val,cdmInfo.seq_no,
                  cdmInfo.created_by,cdmInfo.creation_date,cdmInfo.last_update_by,cdmInfo.last_update_date,
                  cdmInfo.call_cnt,cdmInfo.remark,cdmInfo.sts_cd).then(response => {
          // commit('')
          resolve(response)
        }).catch(error => {
          reject(error)
        })
      })
    },

    DeleteCdm({ commit },par_id){
      return new Promise((resolve,reject) => {
        deleteCdm(par_id).then(response => {
          // commit('')
          resolve(response)
        }).catch(error => {
          reject(error)
        })
      })
    },





  }

}

export default cdm
